src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -97,7 +97,7 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
                     SafeProcessHandle hCurProcess = SafeProcessHandle.InvalidHandle;
                     try
                     {
-                        hCurProcess = ProcessManager.OpenProcess((int)Interop.Kernel32.GetCurrentProcessId(), Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION, true);
+                        hCurProcess = ProcessManager.OpenProcess(Environment.ProcessId, Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION, true);
```

Comment:
Opening a new process handle for the current process just to call IsWow64Process on it looks like an overkill. Calling `Interop.Kernel32.GetCurrentProcess()` should be enough - it is what we do in the other place where `IsWow64Process` is called.

Reply:
I'd rather not change that in this PR.  I'll open an issue for it.

Reply:
https://github.com/dotnet/runtime/issues/38910